src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -189,8 +189,20 @@ export class SharedCredentialsProvider implements CredentialsProvider {
             logger.verbose(
                 `Profile ${this.profileName} contains ${SHARED_CREDENTIAL_PROPERTIES.CREDENTIAL_PROCESS} - treating as Process Credentials`
             )
+            // Guard against hangs: credential_process will wait forever if the process does not return a result.
+            // We must do this now, because CredentialProviderChain calls refresh on *all* providers without a good
+            // way to timeout any single provider.
+            const provider = new AWS.ProcessCredentials({ profile: this.profileName })
+            const timeoutError = new Error(
+                `Timed out while getting credentials from process for profile "${this.profile}"`
+            )
+
+            await Promise.race([
+                provider.refreshPromise(),
+                new Promise((_, reject) => setTimeout(() => reject(timeoutError), 5000)),
```

Comment:
Users might have a long running credential process. Users might be better served by making the timeout value something configurable.

Reply:
Yeah was just about to say if 5 seconds was too short.

Reply:
5 seconds is too short, but configuration will be confusing. We could instead show a message with a button like "Wait longer"

Reply:
A button is doable.